### ISSUE TITLE
React 18 Upgrade: atoms-rendering

### ISSUE THREAD

**Issue by JamieB-gu** [guardian/dotcom-rendering#7417]
This must be done before we can upgrade AR and DCR. An upgrade guide is available here: https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html

**Comment by georgeblahblah** [comment]
It looks like `source-react-components` is still on React 17. We probably need it to be upgraded to React 18 first before upgrading `atoms-rendering`.

**Comment by georgeblahblah** [comment]
Because `react` is a peer dep of `source-react-components`, we can upgrade React independently, even though we will get a mismatched peer dependency warning

**Comment by georgeblahblah** [comment]
I was hunting for prior art and saw @joecowton1 has already done this in https://github.com/guardian/csnx/pull/456! We can close this issue when that PR is merged. Thanks Joe!
